crates/bevy_asset/src/io/memory.rs

```diff
@@ -236,6 +237,46 @@ impl AsyncRead for DataReader {
     }
 }
 
+impl AsyncSeek for DataReader {
+    fn poll_seek(
+        mut self: Pin<&mut Self>,
+        _cx: &mut std::task::Context<'_>,
+        pos: SeekFrom,
+    ) -> Poll<std::io::Result<u64>> {
+        let result = match pos {
+            SeekFrom::Start(offset) => offset.try_into(),
+            SeekFrom::End(offset) => self
+                .data
+                .value()
+                .len()
+                .try_into()
+                .map(|len: i64| len - offset),
+            SeekFrom::Current(offset) => self
+                .bytes_read
+                .try_into()
+                .map(|bytes_read: i64| bytes_read + offset),
+        };
+
+        if let Ok(new_pos) = result {
+            if new_pos < 0 {
```

Comment:
Should there also be a check to ensure `new_pos` is before the end of the `Reader`? I can see the `AsyncRead` implementation is already checking so I don't think this is a safety issue, just want to flag it just in case it's a mistake.